webpack/assets/javascripts/react_app/components/Layout/LayoutActions.js

```diff
@@ -57,3 +62,24 @@ export const changeLocation = location => dispatch => {
     },
   });
 };
+
+export const onLogout = (url = LAYOUT_LOGOUT_URL) => async dispatch => {
+  dispatch(showLoading());
+  const onResultsSuccess = response =>
+    dispatch({
+      type: LAYOUT_LOGOUT,
+    });
+
+  const onResultsError = response => {
+    dispatch(hideLoading());
+    const message = __(response.toString());
+    notify({ message, type: 'error' });
+  };
+
+  try {
+    const response = await API.post(url);
```

Comment:
I think this doesn't follow the redirect? afair post will send a redirect after it log you out.

Reply:
the post returns a response a long time before it actually redirects you. thats why i'm dispatching a spinner when the action starts and I don't disable it.

Reply:
my point is that the server respond back with a redirect (check the POST result) - my assumption is that you need to follow the redirect in code now?